addons/stock_landed_costs/models/stock_landed_costs.py

```diff
@@ -0,0 +1,283 @@
+            else:
+                costcor[valuation_line.cost_line_id] = valuation_line.additional_landed_cost
+            tot += valuation_line.additional_landed_cost
+
+        prec = self.env['decimal.precision'].precision_get('Account')
+        # float_compare returns 0 for equal amounts
+        res = not bool(float_compare(tot, landed_cost.amount_total, precision_digits=prec))
+        for costl in costcor.keys():
+            if float_compare(costcor[costl], costl.price_unit, precision_digits=prec):
+                res = False
+        return res
+
+    @api.multi
+    def button_validate(self):
+        self.ensure_one()
+        for cost in self:
+            if cost.state != 'draft':
+                raise UserError(_('Only draft landed costs can be validated'))
+            if not cost.valuation_adjustment_lines or not self._check_sum(cost):
+                raise UserError(_('You cannot validate a landed cost which has no valid valuation adjustments lines. Did you click on Compute?'))
+            move_id = self._create_account_move()
+            quant_dict = {}
+
+            for line in cost.valuation_adjustment_lines:
+                if not line.move_id:
+                    continue
+                per_unit = line.final_cost / line.quantity
+                diff = per_unit - line.former_cost_per_unit
+                quants = [quant for quant in line.move_id.quant_ids]
+                for quant in quants:
+                    if quant.id not in quant_dict:
+                        quant_dict[quant.id] = quant.cost + diff
+                    else:
+                        quant_dict[quant.id] += diff
+                for key, value in quant_dict.items():
+                    self.env['stock.quant'].sudo().browse(key).write({'cost': value})
+                qty_out = 0
+                for quant in line.move_id.quant_ids:
+                    if quant.location_id.usage != 'internal':
+                        qty_out += quant.qty
+                self._create_accounting_entries(line, move_id, qty_out)
+            self.browse(cost.id).write({'state': 'done', 'account_move_id': move_id.id})
+            move_id.post()
+        return True
+
+    @api.multi
+    def button_cancel(self):
+        for cost in self:
+            if cost.state == 'done':
+                raise UserError(_('Validated landed costs cannot be cancelled, '
+                                'but you could create negative landed costs to reverse them'))
+            cost.write({'state': 'cancel'})
+        return True
+
+    @api.multi
+    def unlink(self):
+        # cancel or raise first
+        self.button_cancel()
+        return super(StockLandedCost, self).unlink()
+
+    @api.multi
+    def compute_landed_cost(self):
+        self.ensure_one()
+        StockValuationLine = self.env['stock.valuation.adjustment.lines']
+        unlink_ids = StockValuationLine.search([('cost_id', 'in', self.ids)])
+        unlink_ids.unlink()
```

Comment:
Use self.valuation_adjustment_lines.unlink() instead of search adjustment lines.